### PR TITLE
Make power grid check event force APCs off

### PR DESCRIPTION
**Pull request by Slava0135**
## About the PR

**Media**

Peek 2023-07-10 19-17

**Changelog**

:cl:
- tweak: now during power grid check event affected APCs can't be turned back on

### CODE REVIEW

**Comment by deltanedas** [comment]
does this affect taking out the board

also this requires not doing the event on salv shuttle and such, the event says station not ship

**Comment by Slava0135** [comment]
I dunno why event now affects ships, i previously fixed this

File: Content.Server/StationEvents/Events/PowerGridCheckRule.cs [hunk 1]
```diff
@@ -19,17 +17,25 @@ public sealed class PowerGridCheckRule : StationEventSystem<PowerGridCheckRuleCo
...
+
         protected override void Started(EntityUid uid, PowerGridCheckRuleComponent component, GameRuleComponent gameRule, GameRuleStartedEvent args)
         {
             base.Started(uid, component, gameRule, args);
 
             if (!TryGetRandomStation(out var chosenStation))
                 return;
 
-            foreach (var (apc, transform) in EntityQuery<ApcComponent, TransformComponent>(true))
+            var query = EntityQueryEnumerator<ApcComponent, TransformComponent>();
```
Comment (metalgearsloth) [inline Content.Server/StationEvents/Events/PowerGridCheckRule.cs:34]: AllEntityQuery

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

File: Content.Server/StationEvents/Events/PowerGridCheckRule.cs (hunk 1 above)
Comment (Slava0135) [reply Content.Server/StationEvents/Events/PowerGridCheckRule.cs:34]: why should it be applied to paused entities?

**Review by AJCM-git — APPROVED** [review]

**Comment by VasilisThePikachu** [comment]
Does this close #16168?

**Comment by Slava0135** [comment]
i guess

**Pull request merged**